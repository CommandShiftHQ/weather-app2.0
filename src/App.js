import React, { useState, useEffect } from "react";
import "./styles/App.css";
import getForecasts from "./requests/getForecasts";
import LocationDetails from "./components/LocationDetails";
import ForecastSummaries from "./components/ForecastSummaries";
import SearchForm from "./components/SearchForm";
import ForecastDetails from "./components/ForecastDetails";

function App() {
  const [selectedDate, setSelectedDate] = useState(0);
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({
    city: null,
    country: null,
  });
  const [query, setQuery] = useState("");

  useEffect(() => {
    getForecasts(setSelectedDate, setForecasts, setLocation);
  }, []);

  const handleForecastSelect = (e) => {
    setSelectedDate(parseInt(e.target.value, 10));
  };

  const handleCitySubmit = () => {
    getForecasts(setSelectedDate, query).then(response => {
      // use setState functions here, instead of injecting them into the service layer
    })
  };

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const { city, country } = location;

  return (
    <div className="App">
      <LocationDetails city={city} country={country} />
      <SearchForm
        handleCitySubmit={handleCitySubmit}
        query={query}
        setQuery={setQuery}
      />
      <ForecastSummaries
        forecasts={forecasts}
        handleForecastSelect={handleForecastSelect}
      />
      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;
