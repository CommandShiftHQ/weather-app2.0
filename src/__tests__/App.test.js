import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "../App";

afterEach(cleanup);

describe("App", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
