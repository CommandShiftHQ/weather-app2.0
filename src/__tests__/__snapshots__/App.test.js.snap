// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      , 
    </h1>
    <input
      class="searchCity"
      placeholder="Search city..."
      type="text"
      value=""
    />
    <br />
    <button
      class="searchButton"
    >
      Search
    </button>
    <div
      class="ForecastSummaries"
    />
  </div>
</DocumentFragment>
`;
